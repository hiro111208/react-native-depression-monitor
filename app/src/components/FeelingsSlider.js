import React, { useState, useEffect } from "react";
import { View, StyleSheet } from "react-native";
import MarkSlider from 'react-native-mark-slider';

const marks = [
  {
    value: 1,
    label: "Not at all",
  },
  {
    value: 2,
    label: "",
  },
  {
    value: 3,
    label: "Somewhat",
  },
  {
    value: 4,
    label: "",
  },
  {
    value: 5,
    label: "Very much",
  },
];
export default function FeelingsSlider(props) {
  const [selectedValue, setSelectValue] = useState("");
  const { setFeelingState, feelingName, feelingState } = props;

  const onChange = (value) => {
    setSelectValue(value);
    setFeelingState({
      ...feelingState,
      [`${feelingName}`]: selectedValue,
    });
  };
  return (
    <View style={styles.root}>
        <MarkSlider
          style={styles.sliderStyle}
          step={25}
          max={100}
          marks={marks}
          onChange={value => onChange({ value })}
        />
    </View>
  );
}

const styles = StyleSheet.create({
  sliderStyle: {
    fontSize: 8,
    color: 'black',
  },
  root: {
    justifyContent: 'center',
    width: '100%'
  },
});
