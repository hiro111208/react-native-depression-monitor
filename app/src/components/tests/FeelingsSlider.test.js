import React from "react";
import { shallow } from "enzyme";
import { render } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import FeelingsSlider from "../FeelingsSlider";

describe("Testing FeelingsSlider.js", () => {
    
    it("FeelingsSlider renders correctly", () => {
      const slider = shallow(<FeelingsSlider />);
      expect(slider.length).toBe(1);
    });
  
    it("FeelingsSlider renders correctly across screens", () => {
      const tree = renderer.create(<FeelingsSlider />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    
});

describe("Testing FeelingsSlider.js component interaction", () => {

  it("Moving slider should call state change", () => {
    const mockFn = jest.fn();
    const initialProps = {
      setFeelingState: {mockFn},
    }
    const wrapper = render(<FeelingsSlider {...initialProps} />);
    const slider = wrapper.getByTestId("SLIDER");
    expect(mockFn).not.toHaveBeenCalled();
    slider.props.onSlidingComplete;
  });
});