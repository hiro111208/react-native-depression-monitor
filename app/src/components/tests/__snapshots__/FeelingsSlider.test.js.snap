// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FeelingsSlider.js FeelingsSlider renders correctly across screens 1`] = `
<View
  style={
    Object {
      "width": "85%",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "marginHorizontal": 16,
          "marginTop": 12,
        },
        Object {
          "marginBottom": 44,
        },
      ]
    }
  >
    <RCTSlider
      disabled={false}
      enabled={true}
      maximumTrackTintColor="#ddd"
      maximumValue={5}
      minimumTrackTintColor="#ffd390"
      minimumValue={1}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onSlidingComplete={[Function]}
      onStartShouldSetResponder={[Function]}
      onValueChange={[Function]}
      step={1}
      style={
        Object {
          "height": 40,
        }
      }
      testID="SLIDER"
      thumbTintColor="#ffa351ff"
      value={0.5}
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 44,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "position": "absolute",
              "textAlign": "center",
            },
            Object {
              "color": "#ddd",
              "transform": Array [
                Object {
                  "translateX": -0,
                },
              ],
              "width": 0,
            },
            Object {
              "left": 5,
            },
          ]
        }
      >
        Not at all
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "position": "absolute",
              "textAlign": "center",
            },
            Object {
              "color": "#ddd",
              "transform": Array [
                Object {
                  "translateX": -0,
                },
              ],
              "width": 0,
            },
            Object {
              "left": 0,
            },
          ]
        }
      >
        Somewhat
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "position": "absolute",
              "textAlign": "center",
            },
            Object {
              "color": "#ddd",
              "transform": Array [
                Object {
                  "translateX": -0,
                },
              ],
              "width": 0,
            },
            Object {
              "left": -5,
            },
          ]
        }
      >
        Very much
      </Text>
    </View>
  </View>
</View>
`;
